tests/cases/compiler/genericTupleWithSimplifiableElements.ts

```diff
@@ -0,0 +1,18 @@
+// @strict: true
+// @noEmit: true
+
+// repro from #52354
+
+type SS1 = string;
+let y: [t: "AAA", ...args: { [S in SS1]: [a: number]; }[SS1]] = ["AAA", 1];
+
+type SS2 = "1" | "2" | "3";
+let z: [t: "AAA", ...args: { [S in SS2]: [a: number]; }[SS2]] = ["AAA", 1];
+
+class I<SS extends string>{
+    f() {
+        let w: [...args: { [S in SS]: [a: number]; }[SS]] = [1];
```

Comment:
this was already working because there is a carve-out for single-element generic tuples here:
https://github.dev/microsoft/TypeScript/blob/104b25d7d505be8a6ebefc9c2a2720778f58c5a6/src/compiler/checker.ts#L21075-L21080